File: JanperCab/5-JanperCab/ClientApp/src/app/_models/DuraformProcess/DuraformProcessPressingDto.ts
import { MachinePresserDto } from './../machine/MachinePresserDto';
import { DuraformProcessDto } from './DuraformProcessDto';

export class DuraformProcessPressingDto extends DuraformProcessDto {
  machineId: number;

  machinePresser: MachinePresserDto;

  getStatus(): string {
    status = this.startTime && this.endTime ? 'Pressed' : 'Pressing';

    if (this.machinePresser) {
      status += ` - ${this.machinePresser.name}`;
    }

    return status;
  }
}

File: JanperCab/3-Application/Dtos/Process/DuraformProcessPressingDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class DuraformProcessPressingDto : DuraformProcessDto
    {
        public int? MachineId { get; set; }

        public MachinePresserDto MachinePresser { get; set; }
    }
}